### PR TITLE
feat: iam engine option

### PR DESCRIPTION
**Pull request by jor2**
### Description

Expose ability to configure IAM engine in SM instance in the DA.

https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/issues/77

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Expose ability to configure IAM engine in SM instance in the DA.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/standard/variables.tf [hunk 1]
```diff
@@ -62,6 +62,18 @@ variable "secret_manager_tags" {
   default     = []
 }
 
+variable "iam_engine_enabled" {
+  type        = bool
+  description = "Set this to true to to configure an IBM Secrets Manager IAM credentials engine for an existing IBM Secrets Manager instance. If set to false, no iam engine will be configured for your secrets manager instance. For more details, see https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-configure-iam-engine."
```
Comment (ocofaigh) [inline solutions/standard/variables.tf:67]: Maybe don't say `for an existing IBM Secrets Manager instance` - the instance is provisioned as part of the DA solution

File: solutions/standard/main.tf [hunk 2]
```diff
@@ -69,3 +69,18 @@ module "secrets_manager" {
   skip_en_iam_authorization_policy = var.skip_en_iam_authorization_policy
   endpoint_type                    = var.service_endpoints == "private" ? var.service_endpoints : "public"
 }
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_iam_secrets_engine = var.iam_engine_enabled && var.iam_engine_name == null ? tobool("When setting var.iam_engine_enabled to true, a value must be passed for var.iam_engine_name") : true
```
Comment (ocofaigh) [inline solutions/standard/main.tf:75]: hmm, not really needed since `iam_engine_name` has a default value.

File: tests/pr_test.go [hunk 3]
```diff
@@ -130,10 +131,12 @@ func TestRunDASolutionSchematics(t *testing.T) {
 		{Name: "ibmcloud_api_key", Value: options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], DataType: "string", Secure: true},
 		{Name: "region", Value: options.Region, DataType: "string"},
 		{Name: "resource_group_name", Value: options.Prefix, DataType: "string"},
+		{Name: "prefix", Value: options.Prefix, DataType: "string"},
```
Comment (ocofaigh) [inline tests/pr_test.go:134]: There is no prefix variable for the DA

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments - also diagram for DA will need to be updated to show IAM engine

**Comment by ocofaigh** [comment]
@jor2 I think there is a bug in the secrets group module that needs to be fixed first -> https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager-secret-group/issues/148

**Comment by jor2** [comment]
still getting
````
 2024/04/10 17:10:57 Terraform apply | Error: CreateSecretGroupWithContext failed Post "https://e9465a83-5a63-4fa9-834e-ba7a86c999e9.us-south.secrets-manager.appdomain.cloud/api/v2/secret_groups": dial tcp: lookup e9465a83-5a63-4fa9-834e-ba7a86c999e9.us-south.secrets-manager.appdomain.cloud on 172.21.0.10:53: no such host
 2024/04/10 17:10:57 Terraform apply | null
 2024/04/10 17:10:57 Terraform apply | 
 2024/04/10 17:10:57 Terraform apply | 
 2024/04/10 17:10:57 Terraform apply |   with module.iam_secrets_engine[0].module.secrets_manager_group_acct[0].ibm_sm_secret_group.secret_group,
 2024/04/10 17:10:57 Terraform apply |   on .terraform/modules/iam_secrets_engine.secrets_manager_group_acct/main.tf line 8, in resource "ibm_sm_secret_group" "secret_group":
 2024/04/10 17:10:57 Terraform apply |    8: resource "ibm_sm_secret_group" "secret_group" {
````

**Comment by jor2** [comment]
> see comments - also diagram for DA will need to be updated to show IAM engine

Updated the diagram, not entirely sure how I should have shown IAM engine.

**Pull request merged**